### ISSUE TITLE
How to get the np files for training?

### ISSUE THREAD

**Issue by htZhang25** [zztant/MGANs#1]
Hello,Thank you for your work.I quite don't know how to prepare my np files as you said"Please prepare your training data as np files"in Readme file.   I'm confused is this np file just the .mat file which should be loaded when I run the gait.py?

**Comment by zztant** [comment]
I am sorry for some negligence in the README, you only need to prepare the .mat file.
For example in CASIA-B，
'data' is the PEIs of all sequences in the CASIA-B.
'view' is view information (1~11)  of all sequences.
'label' is the identity information (1~124)  of all sequences.
'cov' is the walking condition (1~10) of all sequences.
The meaning of these numbers can be found in the .ipynb file.

**Comment by zztant** [comment]
> Hello,Thank you for your work.I quite don't know how to prepare my np files as you said"Please prepare your training data as np files"in Readme file. I'm confused is this np file just the .mat file which should be loaded when I run the gait.py?

I am sorry for some negligence in the README, you only need to prepare the .mat file.
For example in CASIA-B，
'data' is the PEIs of all sequences in the CASIA-B.
'view' is view information (1-11) of all sequences.
'label' is the identity information (1-124) of all sequences.
'cov' is the walking condition (1-10) of all sequences.
The meaning of these numbers can be found in the .ipynb file.

**Comment by htZhang25** [comment]
你好，
      谢谢你的回复。
      我的理解是：
     1我把MATLAB中运行demoPEI的结果pei变量重命名为为data，
     2 再在MATLAB里创建view label cov三个变量，它们的内容矩阵就只是一个数字，接着 
     3  将data view label cov四个变量在MATLAB中存为一个.mat文件
    那么，每一个步态序列文件夹（也就是一个subject的一种情况，一个行走样本）会对应一个.mat文件，是这样吗？期望回答。

**Comment by htZhang25** [comment]
你好，我想我应该这么问，data的维度是明确的，5*64*64,但我不清楚label view cov的维度应该是怎样的,.ipynb文件中load的.mat文件中应该包含我所有要训练的多次行走的数据吧？

**Comment by htZhang25** [comment]
Sorry,I'm still confused about how to get the right .mat file. Should the .mat file contain all datas produced by different gait sequences folders (for example 124 subjects*11 views *4 covs , just 5456 training examples)？Can you tell me some details about how to produce the .mat file ,such as what  the dimention of view \label\ cov should be？I will be waiting for your reply sincerely.

**Comment by zztant** [comment]
> Sorry,I'm still confused about how to get the right .mat file. Should the .mat file contain all datas produced by different gait sequences folders (for example 124 subjects*11 views *4 covs , just 5456 training examples)？Can you tell me some details about how to produce the .mat file ,such as what the dimention of view \label\ cov should be？I will be waiting for your reply sincerely.

In the CASIA-B dataset, the number of sequence is about 124x11x10=13640. Therefore, the dimension of 'label', 'view' and 'cov' is [13640], and the dimension of 'data' is [13640, channel_num, height, width].

I'm sorry for making you confused. I didn't understand how to write code elegantly in deep learning project when I was doing this project. I strongly recommend you to rewrite the code of loading data by using the PyTorch API such as 'torch.utils.data.Dataset' and 'torch.utils.data.DataLoader'.

**Comment by use78908** [comment]
您好:
        我目前在閱讀您的論文以及嘗試將論文中的方法實做出來，目前遇到了一些問題例如使用matlab執行demoPEI.m後生成的PEI影像(5X64X64)內容都是NaN或是0，會無法對應到論文中的Fig.3的5-channel PEI。
還有一些論文上的問題，是否可以透過信箱來向您請教。
謝謝
--------- ---------------
use78908@gmail.com
